src/components/TasksList.tsx

```diff
@@ -1,29 +1,72 @@
+import { Button as MaterialButton } from '@mui/material';
 
-
-function TasksList( props: {items: Item[], onDelete: (id: number) => void, clearAll: () => void}) {
-        { 
-            return (
-                <div className={styles.tasks}>
+function TasksList(props: {
+    item: string;
+    itemsArr: Item[];
+    editedTaskId?: number;
+    onDelete: (id: number) => void;
+    onEdit: (id: number) => void;
+    onSave: (id: number, taskToSave: string) => void;
+    clearAll: () => void;
+    typeTextToEdit: (event: ChangeEvent<HTMLInputElement>) => void;
+}) {
+    {
+        return (
+            <div className={styles.tasks}>
                 <ul className={styles['ul--tasks--list']}>
-                    {
-                        (props.items.length !== 0)? 
-                            props.items.map((task) =>
+                    {props.itemsArr.length !== 0 ? (
+                        props.itemsArr.map((task) => (
                             <li data-task={task.item} className={styles['list--item']} key={task.id}>
-                                <h3 data-task-title={task.item} title={task.item} className={styles['list--added-task']} >{task.item}</h3>
-                                <button className={styles['delete--task-button']} onClick={() => props.onDelete(task.id)}>
-                                    <img src="../src/icons/delete.png" className={styles['delete--task-button']} height="20px" alt="" />Delete</button>
-                            </li>                             
-                        ) : <h3>List is empty!</h3>
-                    } 
+                                {props.editedTaskId !== task.id ? (
+                                    <>
+                                        <h3 data-task-title={task.item} title={task.item} className={styles['list--added-task']}>
+                                            {task.item}
+                                        </h3>
+                                        <button className={styles['delete--task-button']} onClick={() => props.onDelete(task.id)}>
+                                            <img src="../src/icons/delete.png" className={styles['icon--task-button']} />
+                                            Delete
+                                        </button>
+                                        <button className={styles['edit--task-button']} onClick={() => props.onEdit(task.id)}>
+                                            <img src="../src/icons/edit.png" className={styles['icon--task-button']} />
+                                            Edit
+                                        </button>
+                                    </>
+                                ) : (
+                                    <>
+                                        <input type="text" name="edit-task-text" placeholder={task.item} onChange={props.typeTextToEdit} className={styles['list--added-task']} value={props.item} />
```

Comment:
I suggest you rename the prop for the function passed to this input's onChange property to "onChange". Naming it differently makes it confusing as the sole purpose of this function is to be passed along to that input inside.

```suggestion
                                        <input type="text" name="edit-task-text" placeholder={task.item} onChange={props.onChange} className={styles['list--added-task']} value={props.item} />
```